### PR TITLE
feat(service-helm): update networking modal to add service and port selection

### PR DESCRIPTION
**Pull request by ctjhoa**
# What does this PR do?

https://qovery.atlassian.net/browse/FRT-1261

2024-09-09-110801_1920x1080_scrot

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Truth can only be found in one place: the code

File: libs/domains/service-helm/feature/src/lib/hooks/use-kubernetes-services/use-kubernetes-services.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { useQuery } from '@tanstack/react-query'
+import { queries } from '@qovery/state/util-queries'
+
+export interface UseKubernetesServicesProps {
+  helmId: string
+}
+
+export function useKubernetesServices({ helmId }: UseKubernetesServicesProps) {
```
Comment (RemiBonnet) [inline libs/domains/service-helm/feature/src/lib/hooks/use-kubernetes-services/use-kubernetes-services.tsx:8]: Perhaps it should be better to have `Services` before? Like we have in the domains `service`? 

`useServicesKubernetes`

File: libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx [hunk 2]
```diff
@@ -1,33 +1,48 @@
...
+      internal_port: `${port?.internal_port ?? ''}`,
+      external_port: `${port?.external_port ?? '443'}`,
       service_name: port?.service_name ?? '',
       namespace: port?.namespace ?? '',
       protocol: port ? port.protocol : PortProtocolEnum.HTTP,
       name: port ? port.name : '',
     },
     mode: 'all',
   })
+  const { data: kubernetesServices, isLoading } = useKubernetesServices({ helmId })
```
Comment (RemiBonnet) [inline libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx:27]:
```suggestion
  const { data: kubernetesServices = [], isLoading } = useKubernetesServices({ helmId })
```

File: libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx [hunk 3]
```diff
@@ -1,33 +1,48 @@
...
 
   const { control, watch, setValue } = methods
 
   const watchInternalPort = watch('internal_port')
   const watchServiceName = watch('service_name')
 
+  const availableServiceNames = kubernetesServices?.map(({ metadata: { name } }) => ({
```
Comment (RemiBonnet) [inline libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx:34]:
```suggestion
  const availableServiceNames = kubernetesServices.map(({ metadata: { name } }) => ({
```

File: libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx [hunk 4]
```diff
@@ -76,71 +91,118 @@ export function NetworkingPortSettingModal({ port, onClose, onSubmit }: Networki
...
-                name={field.name}
-                onChange={(e: FormEvent<HTMLInputElement>) => {
-                  const name = `p${watchInternalPort}-${e.currentTarget.value}`
-                  setValue('name', name.slice(0, namePatternRules.maxLength.value))
-                  field.onChange(e)
+          {isLoading ? (
+            <div className="flex justify-center">
+              <LoaderSpinner />
+            </div>
+          ) : availableServiceNames && availableServiceNames?.length > 0 ? (
```
Comment (RemiBonnet) [inline libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx:98]:
```suggestion
          ) : availableServiceNames.length > 0 ? (
```

File: libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx [hunk 5]
```diff
@@ -76,71 +91,118 @@ export function NetworkingPortSettingModal({ port, onClose, onSubmit }: Networki
...
+                  label="Service name"
+                  value={field.value}
+                  options={availableServiceNames}
+                  error={error?.message}
+                  onChange={(e) => {
+                    const name = `p${watchInternalPort}-${e}`
+                    setValue('name', name.slice(0, namePatternRules.maxLength.value))
+                    field.onChange(e)
+                  }}
+                  portal
```
Comment (RemiBonnet) [inline libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx:117]:
```suggestion
                  portal
                  isSearchable
```

File: libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx [hunk 6]
```diff
@@ -76,71 +91,118 @@ export function NetworkingPortSettingModal({ port, onClose, onSubmit }: Networki
...
+                    label="Service port"
+                    value={field.value}
+                    options={availableServicePorts}
+                    error={error?.message}
+                    onChange={(e) => {
+                      const name = `p${e}-${watchServiceName}`
+                      setValue('name', name.slice(0, namePatternRules.maxLength.value))
+                      field.onChange(e)
+                    }}
+                    portal
```
Comment (RemiBonnet) [inline libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx:168]:
```suggestion
                    portal
                    isSearchable
```

File: libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx [hunk 7]
```diff
@@ -169,6 +231,20 @@ export function NetworkingPortSettingModal({ port, onClose, onSubmit }: Networki
               />
             )}
           />
+          <Controller
+            name="namespace"
+            defaultValue=""
```
Comment (RemiBonnet) [inline libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx:236]: Is it necessary to set an `defaultValue` empty for each controller?

File: libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx [hunk 8]
```diff
@@ -76,71 +91,118 @@ export function NetworkingPortSettingModal({ port, onClose, onSubmit }: Networki
...
+                  }}
+                  value={field.value}
+                  label="Service name"
+                  error={error?.message}
+                />
+              )}
+            />
+          )}
+          {watchServiceName ? (
+            isLoading ? (
```
Comment (RemiBonnet) [inline libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx:145]: I wonder if the best is to have an global wrapper `isLoading` and hide other fields if it's not loaded, wdyt? It avoid space gap when `isLoading` is finally false

**Review by RemiBonnet — CHANGES_REQUESTED** [review]
Thanks @ctjhoa, I added some feedback and comments!

File: libs/domains/service-helm/feature/src/lib/hooks/use-kubernetes-services/use-kubernetes-services.tsx (hunk 1 above)
Comment (ctjhoa) [reply libs/domains/service-helm/feature/src/lib/hooks/use-kubernetes-services/use-kubernetes-services.tsx:8]: Nope this is the correct english spelling for what it represents.
Here we list services with a kubernetes "attribute"

File: libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx (hunk 8 above)
Comment (ctjhoa) [reply libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx]: I'm not really sensitive to that.
I've made the change
https://github.com/Qovery/console/pull/1654/commits/a37dfefcad1b8c178aefaff7c4fea957cf935259
tell me if you like it better.

File: libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx (hunk 7 above)
Comment (ctjhoa) [reply libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx]: IMO this was useless, I remove them
https://github.com/Qovery/console/pull/1654/commits/dfc81efd64629e7d3d8219efc4c24d41535403b4

File: libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx (hunk 6 above)
Comment (ctjhoa) [reply libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx:212]: Fixed https://github.com/Qovery/console/pull/1654/commits/dfc81efd64629e7d3d8219efc4c24d41535403b4

File: libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx (hunk 5 above)
Comment (ctjhoa) [reply libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx]: Fixed https://github.com/Qovery/console/pull/1654/commits/dfc81efd64629e7d3d8219efc4c24d41535403b4

File: libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx (hunk 4 above)
Comment (ctjhoa) [reply libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx]: Fixed https://github.com/Qovery/console/pull/1654/commits/dfc81efd64629e7d3d8219efc4c24d41535403b4

File: libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx (hunk 3 above)
Comment (ctjhoa) [reply libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx]: Fixed https://github.com/Qovery/console/pull/1654/commits/dfc81efd64629e7d3d8219efc4c24d41535403b4

File: libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx (hunk 2 above)
Comment (ctjhoa) [reply libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx]: Fixed https://github.com/Qovery/console/pull/1654/commits/dfc81efd64629e7d3d8219efc4c24d41535403b4

**Review by RemiBonnet — CHANGES_REQUESTED** [review]
Thanks @ctjhoa 
I just see I can create a port without `service port`, the validation isn't correct if I change my `service name` after selected one. Could you fix it? 
Screenshot 2024-09-10 at 09 02 09




Plus, we displayed the `namespace` even if we don't have value, I know it's not related to your PR but can you hide it?
Screenshot 2024-09-10 at 09 00 17

**Review by RemiBonnet — APPROVED** [review]

**Pull request merged**